src/Mod/Material/Gui/AppearancePreview.cpp

```diff
@@ -148,78 +191,118 @@ void AppearancePreview::applySettings()
     viewSettings->applySettings();
 }
 
+void AppearancePreview::setCoinTexture()
+{
+    _switch->whichChild.setValue(1);
+}
+
+void AppearancePreview::setCoinMaterial()
+{
+    _switch->whichChild.setValue(0);
+}
+
 void AppearancePreview::setAmbientColor(const QColor& color)
 {
+    setCoinMaterial();
     _material->ambientColor.setValue(
         SbColor(color.red() / 255.0, color.green() / 255.0, color.blue() / 255.0));
     _material->ambientColor.setDefault(false);
 }
 
 void AppearancePreview::setDiffuseColor(const QColor& color)
 {
+    setCoinMaterial();
     _material->diffuseColor.setValue(
         SbColor(color.red() / 255.0, color.green() / 255.0, color.blue() / 255.0));
     _material->diffuseColor.setDefault(false);
 }
 
 void AppearancePreview::setSpecularColor(const QColor& color)
 {
+    setCoinMaterial();
     _material->specularColor.setValue(
         SbColor(color.red() / 255.0, color.green() / 255.0, color.blue() / 255.0));
     _material->specularColor.setDefault(false);
 }
 
 void AppearancePreview::setEmissiveColor(const QColor& color)
 {
+    setCoinMaterial();
     _material->emissiveColor.setValue(
         SbColor(color.red() / 255.0, color.green() / 255.0, color.blue() / 255.0));
     _material->emissiveColor.setDefault(false);
 }
 
 void AppearancePreview::setShininess(double value)
 {
+    setCoinMaterial();
     _material->shininess.setValue(value);
     _material->shininess.setDefault(false);
 }
 
 void AppearancePreview::setTransparency(double value)
 {
+    setCoinMaterial();
     _material->transparency.setValue(value);
     _material->transparency.setDefault(false);
 }
 
+void AppearancePreview::setTexture(const QImage& image)
+{
+    setCoinTexture();
+
+    SoSFImage texture;
+    Gui::BitmapFactory().convert(image, texture);
+    _texture->image = texture;
+}
+
+void AppearancePreview::setTextureScaling(double scale)
+{}
```

Comment:
```suggestion
{
    (void) scale;
}
```
(or just leave the parameter unnamed here, or use the Boost unused thing).